File: diffx-python/tests/docs_examples/index_examples.py
import unittest
import tempfile
import json
import os
from diffx_python import diffx_python as diffx

class IndexExamplesTest(unittest.TestCase):
    
    def create_temp_json(self, data):
        """Helper to create temporary JSON files"""
        temp_file = tempfile.NamedTempFile(mode='w', suffix='.json', delete=False)
        json.dump(data, temp_file)
        temp_file.close()
        return temp_file.name
    
    def tearDown(self):
        """Clean up temporary files"""
        for filename in getattr(self, '_temp_files', []):
            if os.path.exists(filename):
                os.unlink(filename)
    
    def test_index_semantic_diff(self):
        """Test case 1: diffx config1.json config2.json"""
        file1 = self.create_temp_json({"name": "myapp", "version": "1.0"})
        file2 = self.create_temp_json({"version": "1.1", "name": "myapp"})
        
        self._temp_files = [file1, file2]
        
        result = diffx.diff(file1, file2)
        self.assertIn('version', result)
        self.assertIn('1.0', result)
        self.assertIn('1.1', result)

if __name__ == '__main__':
    unittest.main()

File: diffx-core/tests/docs_examples/index_examples.rs
use diffx_core::*;
use serde_json::{json, Value};

/// Test case 1: diffx config1.json config2.json
#[test]
fn test_index_semantic_diff() {
    let v1 = json!({"name": "myapp", "version": "1.0"});
    let v2 = json!({"version": "1.1", "name": "myapp"});

    let diffs = diff(&v1, &v2, None).unwrap();
    assert_eq!(diffs.len(), 1);
    match &diffs[0] {
        DiffResult::Modified(path, old, new) => {
            assert_eq!(path, "version");
            assert_eq!(old, &json!("1.0"));
            assert_eq!(new, &json!("1.1"));
        }
        _ => panic!("Expected Modified diff result"),
    }
}
